payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/WalletServiceImpl.java

```diff
@@ -22,6 +23,14 @@ public WalletEntity getWalletById(final long id) {
         return walletRepository.findById(id).orElseThrow(NoWalletFoundException::new);
     }
 
+    public WalletEntity getWalletByUser(final long userId, final UserType userType) {
+        final WalletEntity walletEntity = walletRepository.findWalletEntityByUserIdAndUserType(userId, userType);
```

Comment:
If this query is going to be used often, it can be a bottleneck. Can you explain why, and how can we solve this issue ?

Reply:
beacuse searching for wallet by 2 vars takes long time

Reply:
what is the difference between searching by id and any other field ?

Reply:
search by ID performs instantly

Reply:
because it is primary key

Reply:
because it's like index in array

Reply:
What can we do to increase reading speed of other fields ?

Reply:
@kostya4201337 do you understand what you need to do?

Reply:
create index

Reply:
For which field/fields ?

Reply:
user id and user type

Reply:
You are so fast to answer. 

Let's think a bit. When we use index we reduce speed of writing because we need to add this data structure in addition to actual data. Can we improve your idea and use only one index?

For each user id it's possible to have max 2 elements. But for each user type you can have lots of elements. Based on it can we use only one field for index ? And which one?

Reply:
we need to create index for user id because 

> For each it's possible to have max 2 elements.